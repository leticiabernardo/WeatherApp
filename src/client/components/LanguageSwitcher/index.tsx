import { Box, Button } from '@chakra-ui/react';

const LanguageSwitcher = () => {
  return (
    <Box as="span" padding="0 4px 4px" color="#a9a9a9">
      <Button variant="link" color="white">
        EN
      </Button>
      |
      <Button variant="link" color="white">
        PT
      </Button>
    </Box>
  );
};

export default LanguageSwitcher;
