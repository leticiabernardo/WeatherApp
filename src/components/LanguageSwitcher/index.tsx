/* eslint-disable @typescript-eslint/no-floating-promises */
import { Box, Button } from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';

const LanguageSwitcher = () => {
  const { i18n } = useTranslation();

  const handleChangeLanguage = (lang: string) => {
    if (lang === i18n.language) {
      return;
    }
    i18n.changeLanguage(lang);
  };

  return (
    <Box as="span" padding="0 4px 4px" color="#a9a9a9">
      <Button
        variant="link"
        color={i18n.language === 'en_US' ? 'white' : '#999'}
        _hover={{ color: '#ccc' }}
        _focus={{ outline: 'none' }}
        onClick={() => handleChangeLanguage('en_US')}
      >
        EN
      </Button>
      |
      <Button
        variant="link"
        color={i18n.language === 'pt_BR' ? 'white' : '#999'}
        _hover={{ color: '#ccc' }}
        _focus={{ outline: 'none' }}
        onClick={() => handleChangeLanguage('pt_BR')}
      >
        PT
      </Button>
    </Box>
  );
};

export default LanguageSwitcher;
