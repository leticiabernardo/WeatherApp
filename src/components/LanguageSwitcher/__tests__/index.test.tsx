import { render, cleanup, fireEvent } from '@testing-library/react';
import LanguageSwitcher from '../index';

describe('language switcher render', () => {
  afterEach(cleanup);

  test('render the component', () => {
    const { container } = render(<LanguageSwitcher />);
    expect(container).toMatchSnapshot();
  });

  test('should have the button to switch to English language', () => {
    const { getByText } = render(<LanguageSwitcher />);

    expect(getByText('EN')).toBeInTheDocument();
  });

  test('should have the button to switch to Portuguese language', () => {
    const { getByText } = render(<LanguageSwitcher />);

    expect(getByText('PT')).toBeInTheDocument();
  });
});

describe('language switcher events', () => {
  afterEach(cleanup);

  test('should change to Portuguese language', () => {
    const { container, getByText } = render(<LanguageSwitcher />);
    const button = getByText('PT');
    fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });

  test('should change to English language', () => {
    const { container, getByText } = render(<LanguageSwitcher />);
    const button = getByText('EN');
    fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });

  test('should change languages', () => {
    const { getByText } = render(<LanguageSwitcher />);
    const buttonPT = getByText('PT');
    const buttonEN = getByText('EN');

    fireEvent.click(buttonPT);
    fireEvent.click(buttonEN);
    fireEvent.click(buttonPT);
  });
});
