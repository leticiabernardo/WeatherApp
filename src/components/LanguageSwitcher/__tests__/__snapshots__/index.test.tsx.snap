// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`language switcher events should change to English language 1`] = `
<div>
  <span
    class="css-xe4d2t"
  >
    <button
      class="chakra-button css-hu7fh3"
      type="button"
    >
      EN
    </button>
    |
    <button
      class="chakra-button css-yqy1ty"
      type="button"
    >
      PT
    </button>
  </span>
</div>
`;

exports[`language switcher events should change to Portuguese language 1`] = `
<div>
  <span
    class="css-xe4d2t"
  >
    <button
      class="chakra-button css-hu7fh3"
      type="button"
    >
      EN
    </button>
    |
    <button
      class="chakra-button css-yqy1ty"
      type="button"
    >
      PT
    </button>
  </span>
</div>
`;

exports[`language switcher render render the component 1`] = `
<div>
  <span
    class="css-xe4d2t"
  >
    <button
      class="chakra-button css-hu7fh3"
      type="button"
    >
      EN
    </button>
    |
    <button
      class="chakra-button css-yqy1ty"
      type="button"
    >
      PT
    </button>
  </span>
</div>
`;
